server/model/pipeline.go

```diff
@@ -20,46 +20,85 @@ import (
+	Created             int64                  `json:"created"             xorm:"'created' NOT NULL DEFAULT 0 created"`
+	Updated             int64                  `json:"updated"             xorm:"'updated' NOT NULL DEFAULT 0 updated"`
+	Started             int64                  `json:"started"             xorm:"started"`
+	Finished            int64                  `json:"finished"            xorm:"finished"`
+	Reviewer            string                 `json:"reviewed_by"         xorm:"reviewer"`
+	Reviewed            int64                  `json:"reviewed"            xorm:"reviewed"`
+	Workflows           []*Workflow            `json:"workflows,omitempty" xorm:"-"`
+	AdditionalVariables map[string]string      `json:"variables,omitempty" xorm:"json 'additional_variables'"`
+
+	// event related
+
+	Event WebhookEvent `json:"event"                       xorm:"event"`
+	// TODO change json to 'commit' in next major
+	Commit          *Commit      `json:"commit_pipeline"             xorm:"json 'commit'"`
+	Branch          string       `json:"branch"                      xorm:"branch"`
+	Ref             string       `json:"ref"                         xorm:"ref"`
+	Refspec         string       `json:"refspec"                     xorm:"refspec"`
+	ForgeURL        string       `json:"forge_url"                   xorm:"forge_url"`
+	Author          string       `json:"author"                      xorm:"author"`
+	Avatar          string       `json:"author_avatar"               xorm:"varchar(500) 'avatar'"`
+	ChangedFiles    []string     `json:"changed_files,omitempty"     xorm:"LONGTEXT 'changed_files'"`
+	Deployment      *Deployment  `json:"deployment,omitempty"        xorm:"json 'deployment'"`
+	IsPrerelease    bool         `json:"is_prerelease,omitempty"     xorm:"is_prerelease"`
+	PullRequest     *PullRequest `json:"pull_request,omitempty"      xorm:"json 'pr'"`
+	Cron            string       `json:"cron,omitempty"              xorm:"cron"`
+	ReleaseTagTitle string       `json:"release_tag_title,omitempty" xorm:"release_tag_title"`
+}
+
+// APIPipeline TODO remove in next major.
+type APIPipeline struct {
+	*Pipeline
+
+	DeployTo   string `json:"deploy_to"`
+	DeployTask string `json:"deploy_task"`
+	Commit     string `json:"commit"`
+
+	Title             string   `json:"title"`
+	Message           string   `json:"message"`
+	Timestamp         int64    `json:"timestamp"`
+	Sender            string   `json:"sender"`
+	Email             string   `json:"author_email"`
+	PullRequestLabels []string `json:"pr_labels,omitempty"`
+	FromFork          bool     `json:"from_fork,omitempty"`
 } //	@name Pipeline
 
 // TableName return database table name for xorm.
 func (Pipeline) TableName() string {
 	return "pipelines"
 }
 
+func (p *Pipeline) ToAPIModel() *APIPipeline {
+	ap := &APIPipeline{
+		Pipeline:  p,
+		Commit:    p.Commit.SHA,
+		Title:     p.Commit.Message,
```

Comment:
This should be using sth like the `pr.title` or `release_tag_title` or `cron` based on the event to be backwards compatible.